crates/astria-conductor/src/celestia/reporting.rs

```diff
@@ -53,7 +53,8 @@ impl<'a> Serialize for ReportReconstructedBlock<'a> {
         ];
         let mut state = serializer.serialize_struct("ReconstructedBlockInfo", FIELDS.len())?;
         state.serialize_field(FIELDS[0], &self.0.celestia_height)?;
-        state.serialize_field(FIELDS[1], &base64(&self.0.block_hash))?;
+        // TODO: use the block hash's Display impl for this
+        state.serialize_field(FIELDS[1], &base64(&*self.0.block_hash))?;
```

Comment:
Is there a reason this can't be done now?
```suggestion
        state.serialize_field(FIELDS[1], &self.0.block_hash.to_string())?;
```

Reply:
Ha, no, thanks for pointing that out!

Reply:
Addressed by  01fdbba2 using an ad-hoc `SerializeDisplay` newtype so that we can use [`Serializer::collect_str`](https://docs.rs/serde/latest/serde/ser/trait.Serializer.html#method.collect_str) and avoid the allocation.